resources/lib/cache.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: UTF-8 -*-
+#
+# Copyright (C) 2020, Team Kodi
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# pylint: disable=missing-docstring
+#
+# This is based on the metadata.tvmaze scrapper by Roman Miroshnychenko aka Roman V.M.
+
+"""Cache-related functionality"""
+
+import os
+import pickle
+import xbmcvfs
+
+from .utils import ADDON, logger
+
+try:
+    from typing import Optional, Text, Dict, Any  # pylint: disable=unused-import
+except ImportError:
+    pass
+
+
+def _get_cache_directory():  # pylint: disable=missing-docstring
+    # type: () -> Text
+    temp_dir = xbmcvfs.translatePath('special://temp')
+    cache_dir = os.path.join(temp_dir, 'scrapers', ADDON.getAddonInfo('id'))
+    if not xbmcvfs.exists(cache_dir):
+        xbmcvfs.mkdir(cache_dir)
+    logger.debug('the cache dir is ' + cache_dir)
+    return cache_dir
+
+
+CACHE_DIR = _get_cache_directory()  # type: Text
+
+
+def cache_show_info(show_id, data, lang, part):
+    # type: (Text, Dict[Text, Any], Text, Text) -> None
+    """
+    Save show_info dict to cache
+    """
+    if not (data and part):
+        return None
+
+    file_name = str(show_id) + '-' + lang + '.pickle'
+    cache = load_show_info_from_cache(show_id, lang)
+    if cache:
+        cache[part] = data
+    else:
+        cache = {
+            part: data
+        }
+    with open(os.path.join(CACHE_DIR, file_name), 'wb') as fo:
+        pickle.dump(cache, fo, protocol=2)
```

Comment:
Pickle-based cache proved to be unreliable across Python versions and even across Python 3 minor versions. This is why I had to switch to JSON-based cache in my TVmaze scraper: https://github.com/romanvm/kodi.tvmaze/blob/master/metadata.tvmaze/libs/cache_service.py
In Python 3 you won't need to do explicit encoding before saving JSON and can just use `json.dump()`.

Reply:
Thanks for looking at this. No part of the caching or fanarttv part is mine; all is stolen. I will go steal yours and update this :D